04_numeronym/alextmz/main_test.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_numeronyms(t *testing.T) {
+	var tests = map[string]struct {
+		arg  []string
+		want []string
+	}{
+		"empty string": {
+			[]string{""},
+			[]string{""}},
+		"single string of 1 element": {
+			[]string{"a"},
+			[]string{"a"}},
+		"single long string": {
+			[]string{"abracadabra"},
+			[]string{"a9a"}},
+		"multiple empty strings": {
+			[]string{"", "", ""},
+			[]string{"", "", ""}},
+		"multiple strings with 1 element": {
+			[]string{"a", "b", "c", "d"},
+			[]string{"a", "b", "c", "d"}},
+		"multiple long strings": {
+			[]string{"abracadabra", "alakazam", "hocuspocus", "mumbojumbo"},
+			[]string{"a9a", "a6m", "h8s", "m8o"}},
+		"mixed strings": {
+			[]string{"pirilimpimpim", "MIA", "", "Zzz", "Zoe", "Pizzaz"},
+			[]string{"p11m", "MIA", "", "Zzz", "Zoe", "P4z"}},
+	}
```

Comment:
You are missing tests with non-ascii unicode runes in your strings. At the beginning, middle and end. If you add these, you'll find your numeronym function fails with strings that start or end with unicode runes represented as more than one byte in utf-8 encoding (the encoding that Go uses for strings).